src/components/ChatPopUp.tsx

```diff
@@ -172,31 +181,44 @@ export function ChatPopUp(props: ChatPopUpProps) {
   );
 
   useEffect(() => {
-    // Open panel on load if openOnLoad prop is true or there are messages in state (from browser storage)
-    if (!renderChat && (openOnLoad || messages.length > 1)) {
+    /* Open panel on load if: 
+      - openOnLoad prop is true or there are messages in state (from browser storage), and local storage flag not set 
+      - local storage flag openOnLoadLocalStorage is set and is true */
+    if (
```

Comment:
consider checking for `window.localStorage.getItem(openOnLoadStateKey)` within useEffect. For onClick/onClose, you don't need to guard check for `openOnLoadLocalStorage`, just directly set the value should be fine